slsim/Deflectors/cluster_catalog_lens.py

```diff
@@ -0,0 +1,281 @@
+            "e1_light" not in members_column_names
+            or "e2_light" not in members_column_names
+        ):
+            members_list["e1_light"] = -np.ones(n_members)
+            members_list["e2_light"] = -np.ones(n_members)
+        if (
+            "e1_mass" not in members_column_names
+            or "e2_mass" not in members_column_names
+        ):
+            members_list["e1_mass"] = -np.ones(n_members)
+            members_list["e2_mass"] = -np.ones(n_members)
+        if "n_sersic" not in members_column_names:
+            members_list["n_sersic"] = -np.ones(n_members)
+        if "gamma_pl" not in members_column_names:
+            members_list["gamma_pl"] = np.ones(n_members) * 2
+
+        self._f_vel_disp = vel_disp_abundance_matching(
+            members_list, z_max=0.5, sky_area=sky_area, cosmo=cosmo
+        )
+
+        self._members_select = object_cut(members_list, **kwargs_cut)
+        self._cluster_select = cluster_list[
+            np.isin(cluster_list["cluster_id"], self._members_select["cluster_id"])
+        ]
+
+        self._members_select["vel_disp"] = self._f_vel_disp(
+            np.log10(self._members_select["stellar_mass"])
+        )
+
+        self._kwargs_mass2light = kwargs_mass2light
+
+        self._num_select = len(self._cluster_select)
+
+        self._cosmo = cosmo
+
+        # TODO: random reshuffle of matched list
+
+    def deflector_number(self):
+        """
+
+        :return: number of deflectors
+        """
+        number = self._num_select
+        return number
+
+    def draw_deflector(self):
+        """
+        :return: dictionary of complete parameterization of deflector
+        """
+        index = random.randint(0, self._num_select - 1)
+        deflector = self.draw_cluster(index)
+        members = self.draw_members(deflector["cluster_id"])
+        deflector["subhalos"] = members
+        return deflector
+
+    def draw_cluster(self, index):
+        cluster = self._cluster_select[index]
+        if cluster["halo_mass"] == -1:
+            cluster["halo_mass"] = mass_richness_relation(
+                cluster["richness"], self.richness_fn
+            )
+        if cluster["concentration"] == -1:
+            cluster["concentration"] = concent_m_w_scatter(
+                np.array([cluster["halo_mass"]]), cluster["z"], sig=0.33
+            )[0]
+        if cluster["vel_disp"] == -1:
+            cluster["vel_disp"] = vel_disp_nfw(
+                cluster["halo_mass"], cluster["concentration"], self.cosmo, cluster["z"]
+            )
+        if cluster["e1_mass"] == -1 or cluster["e2_mass"] == -1:
+            e, phi = gene_e_ang_halo(np.array([cluster["halo_mass"]]))
+            e1, e2 = phi_q2_ellipticity(np.deg2rad(phi[0]), 1 - e[0])
+            cluster["e1_mass"] = e1
+            cluster["e2_mass"] = e2
+        return dict(cluster)
+
+    def draw_members(
+        self, cluster_id, center_scatter=0.2, max_dist_arcsec=80, bcg_band="r"
+    ):
```

Comment:
Missing documentation of input parameters.

Reply:
added documentation